main/src/library/File.flix

```diff
@@ -299,29 +301,57 @@ namespace File {
     /// `count` - the number of bytes read.
     /// `charSet` - the specific charset to be used to decode the bytes.
     ///
-    pub def readWith(opts: {offset :: Int64, count :: Int32, charSet :: String}, f : String): Result[String, String] & Impure = {
+    pub def readWith(opts: {offset :: Int64, count :: Int32, charSet :: String}, f : String): Result[String, String] & Impure =
         try {
             import new java.io.File(String): ##java.io.File & Impure as newFile;
             import new java.io.FileInputStream(##java.io.File): ##java.io.FileInputStream & Impure as newInputStream;
-            import java.io.FileInputStream.skip(Int64): Int64 & Impure;
             import java.io.FileInputStream.read(Array[Int8, Static], Int32, Int32): Int32 & Impure;
             import static java.nio.charset.Charset.forName(String): ##java.nio.charset.Charset & Impure;
             import new java.lang.String(Array[Int8, Static], ##java.nio.charset.Charset): String & Impure as newString;
 
             let javaFile = newFile(f);
             let stream = newInputStream(javaFile);
-            discard skip(stream, opts.offset);
-            let bytes = [0i8; opts.count];
-            discard read(stream, bytes, 0, opts.count);
-            let charSet = forName(opts.charSet);
-            let returnString = newString(bytes, charSet);
-            Ok(returnString)
+            match skip((opts.offset / 10_000i64), opts.offset, stream, f) {
+                case Ok(_)    => {
+                    if (0 <= opts.count) {
```

Comment:
Does it make sense to check for this before even starting the function? So checking as the first thing?

Reply:
I'd say so yes, then you can also return `""` for `count=0`